src/java/org/apache/poi/hssf/usermodel/HSSFSheet.java

```diff
@@ -2659,4 +2660,30 @@ public void setActiveCell(CellAddress address) {
         _sheet.setActiveCellRow(row);
         _sheet.setActiveCellCol(col);
     }
+    private void shiftFormulas(int startRow, int endRow, int n, boolean forRowElseColumnShift){ 
```

Comment:
Per your suggestion, I moved the this code from HSSFSheet#shiftRows into its own method, `private void updateFormulasForRowShift(int startRow, int endRow, int n)`

A column shifting variant of this private method is needed.

The best place for column shifting code in HSSFSheet is right below the rowShift and helper methods.

Reply:
I made common method 
private void updateFormulasForShift(int startIndex, int endIndex, int n, FormulaShifter formulaShifter)
used for both row- and column- shifting, depending on formulaShifter argument.